common/kvstore/batch.go

```diff
@@ -0,0 +1,20 @@
+package kvstore
+
+// Batch is a collection of operations that can be applied atomically to a TableStore.
+type Batch[T any] interface {
+	// Put stores the given key / value pair in the batch, overwriting any existing value for that key.
+	// If nil is passed as the value, a byte slice of length 0 will be stored.
+	Put(key T, value []byte)
+	// Delete removes the key from the batch.
+	Delete(key T)
+	// Apply atomically writes all the key / value pairs in the batch to the database.
+	Apply() error
+	// Size returns the number of operations in the batch.
+	Size() uint32
+}
+
+// BatchOperator is an interface for creating new batches.
+type BatchOperator[T any] interface {
```

Comment:
For simplicity, it looks Batch creation can be a function of TableStore

Reply:
`NewBatch` is also a function on the base `Store` object, but I agree this interface isn't really necessary. Removed.